### ISSUE TITLE
Documentation/types for `freeze()` seem incorrect

### ISSUE THREAD

**Issue by Athanasius** [py2exe/py2exe#152]
I was just sprucing up EDMarketConnector's new `Build-exe-and-msi.py` which invokes `py2exe.freeze()` and noticed PyCharm was unhappy with the types for the `console=` and `windows=` arguments.

We have:
```python
    windows=[
        {
            'dest_base': appname,
            'script': APP,
            'icon_resources': [(0, f'{appname}.ico')],
            'other_resources': [(24, 1, open(f'{appname}.manifest').read())],
        }
    ],
    console=[
        {
            'dest_base': appcmdname,
            'script': APPCMD,
            'other_resources': [(24, 1, open(f'{appcmdname}.manifest').read())],
        }
    ],
```
which works perfectly fine.  PyCharm notes that the declared types of those arguments are `list[str]`, and [the documentation](https://github.com/py2exe/py2exe/blob/master/docs/py2exe.freeze.md) says:
> console (list of str): paths of the Python files that will be frozen as console (CLI) executables.
> windows (list of str): paths of the Python files that will be frozen as windows (GUI) executables.

However, e.g. https://github.com/py2exe/py2exe/blob/master/tests/functional/bundlefiles0_sqlite_test/freeze.py has:
```python
freeze(console=[{ "script": "bundlefiles0_sqlite_test.py"}],
...
```
So this `list[dict[...]]` form does seem to be correct.

I do note that https://github.com/py2exe/py2exe/blob/master/py2exe/__init__.py#L16 doesn't actually declare types, so I can only assume PyCharm is picking this up from the docstring.  `mypy` doesn't complain.

**Comment by albertosottile** [comment]
You are right, the documentation should be amended to `(list of dict)`, I will do that asap. Regarding the use of type hints, up to now that was not really possible due to the piggybacking of `setuptools`. With the `freeze`  API this is instead feasible, however I am not sure how well `typing` can represent this heavily dictionary-based API. I will look into that as soon as I have some spare time (currently rewriting part of the C code for 3.11 and onwards).

**Comment by Athanasius** [comment]
Ah, I wasn't necessarily suggesting adding types throughout, but by all means do *if it's useful to your development process*.

It was just that PyCharm picks up on that docstring and highlights an 'error' which is at best distracting, and at worst could cause a user to waste time trying to track down if they're doing things correctly.

**Comment by urishab** [comment]
@Athanasius thanks I didn't look before I posted, sorry about that.

BTW pycharm recommends that the type definition be updated to something like Union[dict,list] or similar

**Comment by albertosottile** [comment]
I think it's easier to have a discussion here: personally, I would be in favor of dropping the `list` option entirely and making a `dict` with at least a `'script'` keyword the only acceptable argument value, in order to further simplify the API. Would you agree?

**Comment by Athanasius** [comment]
As per my opening of this issue, so long as we can still specify the things in that code snippet, I'm happy.

That *is* more than just `script`.

The `icon_resources` is a convenient way to get the correct icon in the executable, and likewise for using `other_resources` to have a specific executable manifest (which we need in order to tweak some options, e.g. DPI awareness, and using the UTF-8 codepage).  It currently works.

I *guess* that, at least at some point, there is/was support for building more than one executable in each of `windows` and `console` ?  If not then it seems unhelpful to require the enclosing list before getting to the dict that actually specifies things.

**Comment by Athanasius** [comment]
I realise now that your wording was a bit ambiguous.  Presumably you mean "move to a dict, which must have at least `script`".

**Comment by albertosottile** [comment]
> I guess that, at least at some point, there is/was support for building more than one executable in each of windows and console ? If not then it seems unhelpful to require the enclosing list before getting to the dict that actually specifies things.

Epic facepalm... you are right, that's why the list is still there. This is stil supported and used by plenty of people, including... myself :D. 

Forget the proposal, then. I will adapt the documentation to `list of dict` and clarify what keywords these dictionaries need to have (only `'script'` should be mandatory).

**Comment by albertosottile** [comment]
Fixed in 0.12.0.2
